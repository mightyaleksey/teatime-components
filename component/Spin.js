'use strict';

const {Component, PropTypes} = require('react');
const {filterProps, genericName} = require('../lib/util');
const {noop} = require('../lib/dash');
const React = require('react');
const cc = require('classnames');

const cssModules = {
  l: require('../style/spin/spin-l.css'),
  m: require('../style/spin/spin-m.css'),
  s: require('../style/spin/spin-s.css'),
  xs: require('../style/spin/spin-xs.css'),
};

class Spin extends Component {
  css = tokenName => genericName(this.props, tokenName)

  focus = noop

  render() {
    const {className, id} = this.props;
    const {css} = this;

    return (
      <span
        {...filterProps(this.props)}
        className={cc(css('wrapper'), className)}
        id={id}>
        <i className={css('control')}/>
      </span>
    );
  }
}

Spin.defaultProps = {
  size: 'm',
  styles: cssModules,
};

Spin.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  size: PropTypes.oneOf([
    'l',
    'm',
    's',
    'xs',
  ]),
  styles: PropTypes.object,
};

module.exports = Spin;
