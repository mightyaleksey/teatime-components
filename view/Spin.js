'use strict';

const { Component, PropTypes } = require('react');
const { styleName } = require('../tool/className');
const { sanitizeProps } = require('../tool/component');
const React = require('react');

class Spin extends Component {
  render() {
    return (
      <span
        {...sanitizeProps(['styles', 'styleName'], this.props)}
        className={styleName(this.props)}/>
    );
  }
}

Spin.defaultProps = {
  styleName: 'control',
  styles: {},
};

Spin.propTypes = {
  styleName: PropTypes.string,
  styles: PropTypes.shape({
    control: PropTypes.string,
  }),
};

module.exports = Spin;
