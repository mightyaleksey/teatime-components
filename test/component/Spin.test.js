import React from 'react';
import Spin from '../../component/Spin';
import renderer from 'react-test-renderer';

test('Spin renders correctly', () => {
  const tree = renderer.create(
    <Spin
      className='outer'
      id='spins id'/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
