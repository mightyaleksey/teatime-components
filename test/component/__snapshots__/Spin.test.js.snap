exports[`test Spin renders correctly 1`] = `
<span
  className="control outer"
  id="spins id" />
`;
